apps/libexec/restart.sh.in

```diff
@@ -1,4 +1,14 @@
 #!/bin/sh
 
-mon stop
-mon start
+# Fix object config (does nothing if we're not a poller)
+mon oconf poller-fix
+
+# This script might be called from a Naemon worker process.
+# As a result, if we do a stop and start of naemon seperatly, it might happen 
+# that this script is killed before we issue the start command. Hence monitor
+# ends up in a stopped state. Calling restart fixes this issue.
+# However for some reason the service restart command does not exists until
+# we have called service stop of either merlin or naemon. Therefore we first
+# restart merlin with seperate commands, and thereafter restart naemon.
+/sbin/service merlind stop && /sbin/service merlind start
```

Comment:
Would it be necessary to restart the merlind daemon if this strange behaviour hadn't occured?

Reply:
I do not think so (for this patch). However this script, I assume, is used in several places. The behavior before this patch was to restart both merlind and naemon using this script.

Reply:
Or rather, the behavior before was:
stop naemon
stop merlind
start merlind
start naemon

I do not think that the order of the restart make any difference although I must admit I am not 100% certain.